app/controllers/banners_controller.rb

```diff
@@ -53,4 +59,9 @@ def destroy
   def banner_params
     params.require(:banner).permit(:active, :content, :name).merge(user: current_user)
   end
+
+  def deactivate_alternate_active_banner
+    alternate_active_banner = current_organization.banners.where(active: true).where.not(id: @banner.id).first
+    alternate_active_banner&.update(active: false)
```

Comment:
I don't think you need to change anything but I want to point something out for you future benefit.

If the call to `update` fails due to some validation error (unlikely, but there are scenarios where the record could get into a state where that happens) then we will be in a state where `alternate_active_banner` is active and we're also trying to make `@banner` active. Usually, you'd want to handle the possibility of this returning `false`. However I think its fine to ignore it in this case as long as we add back in the `only_one_banner_is_active_per_organization` validation since that will catch the issue.

Reply:
Ah, thanks, that makes sense. Was on the fence about removing that validation but couldn't think of when it would be necessary. Here's a good reason!

Reply:
Yes, ideally you would want to do this in a transaction so that both save/updates will be rolled back if one of them fails.

You could do:

```
  def create
    authorize :application, :admin_or_supervisor?

    @banner = current_organization.banners.build(banner_params)

    Banner.transaction do
      deactivate_alternate_active_banner # and make sure to use `update!` in this method
      @banner.save!
    end

    redirect_to banners_path
  rescue
    render :new
  end
```

and similar for the update method.